app/models/declaration/submissions/Submission.scala

```diff
@@ -50,26 +47,26 @@ case class Submission(
     } else None
   }
 
-  lazy val allSubmissionRequestStatuses: Seq[EnhancedStatus] = {
-    val maybeExistingStatuses = for {
+  lazy val allSubmissionRequestStatuses: Seq[EnhancedStatus] = (
+    for {
       subRequestAction <- actions.filter(_.requestType == SubmissionRequest).headOption
       notificationSummaries <- subRequestAction.notifications
     } yield notificationSummaries.map(_.enhancedStatus)
-    maybeExistingStatuses.getOrElse(Seq.empty[EnhancedStatus])
-  }
+  ).getOrElse(Seq.empty[EnhancedStatus])
 
   lazy val isStatusAcceptedOrReceived: Boolean =
     allSubmissionRequestStatuses.intersect(Seq(GOODS_ARRIVED_MESSAGE, GOODS_ARRIVED, RECEIVED)).nonEmpty
 }
```

Comment:
Do we need this `allSubmissionRequestStatuses` as its only used in the `isStatusAcceptedOrReceived` which we can determine from the `latestEnhancedStatus` field of the submission? :thinking: